src/backend/distributed/connection/connection_management.c

```diff
@@ -812,8 +813,11 @@ FinishConnectionListEstablishment(List *multiConnectionList)
 			 * connectionStart and if passed close all non-finished connections
 			 */
 
-			TimestampTz now = GetCurrentTimestamp();
-			if (TimestampDifferenceExceeds(connectionStart, now, NodeConnectionTimeout))
+			instr_time timeSinceStart;
+			INSTR_TIME_SET_CURRENT(timeSinceStart);
+			INSTR_TIME_SUBTRACT(timeSinceStart, connectionStart);
```

Comment:
Could these 3 lines be wrapped in a function? e.g. MillisecondsPassedSince. The recur in several places.